components/form/UploadImageInput.tsx

```diff
@@ -28,6 +31,10 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
 
         if (user) {
           setUserId(user);
+          if (window.location.pathname === pagePathname) {
+            setIsImageUploaded(true);
+            setIsImageRequired(false);
```

Comment:
This deserves a second look. I only see your new state being handled once, when you are already changing the _isImageUploaded_ state.

If it's mirroring the behaviour of an existing state do we really need it there, or could the change here be just an edit to the name of the original state maybe?

Reply:
I completely removed this setIsImageRequired(false); and used this instead isImageUploaded.